src/test/librbd/cache/test_mock_ReplicatedWriteLog.cc

```diff
@@ -47,14 +46,15 @@ inline ImageCtx *get_image_ctx(MockImageCtx *image_ctx) {
 
 namespace librbd {
 namespace cache {
+namespace pwl {
 
 using ::testing::_;
 using ::testing::DoDefault;
 using ::testing::InSequence;
 using ::testing::Invoke;
 
 struct TestMockCacheReplicatedWriteLog : public TestMockFixture {
-  typedef WriteLogCache<librbd::MockImageCtx> MockReplicatedWriteLog;
+  typedef librbd::cache::pwl::ReplicatedWriteLog<librbd::MockImageCtx> MockReplicatedWriteLog;
```

Comment:
This type should be AbstractWriteLog. I don't think the RWL UT will pass with this reference?

Reply:
No, it must be ReplicatedWriteLog here. As it initialized the class with the following words, eg in Line 180. 
MockReplicatedWriteLog rwl(mock_image_ctx, get_cache_state(mock_image_ctx));

Reply:
Later when you define an interface to create SSDWriteLog or ReplicatedWriteLog, we can update it to :
```c++
typedef librbd::cache::pwl::AbstractWriteLog<librbd::MockImageCtx> MockAbstractWriteLog;

....
MockAbstractWriteLog pwl = new AbstractWriteLog(...., type);  // line 180
```

Reply:
There will not be any more new interfaces. We want to remove existing `WriteLogCache` since it's no longer dependent on `ImageCache` class and we can directly invoke `AbstractWriteLog *m_image_cache = new SSDWriteLog` or `AbstractWriteLog *m_image_cache = new ReplicatedWriteLog` in the `ImageDispatch`.

Btw you cannot create `AbstractWriteLog *m_image_cache = new AbstractWriteLog` since it's an abstract class.